langchain/llms/azureml_endpoint.py

```diff
@@ -0,0 +1,255 @@
+        """
+
+    @abstractmethod
+    def format_response_payload(self, output: bytes) -> OUTPUT_TYPE:
+        """Formats the response body according to the output
+        schema of the model. Returns the data type that is
+        received from the response.
+        """
+
+
+class OSSContentFormatter(ContentFormatterBase[str, str]):
+    """Content handler for LLMs from the OSS catalog."""
+
+    content_type = "application/json"
+    accepts = "application/json"
+
+    def format_request_payload(self, prompt: str, model_kwargs: Dict) -> bytes:
+        input_str = json.dumps(
+            {"inputs": {"input_string": [prompt]}, "parameters": model_kwargs}
+        )
+        return str.encode(input_str)
+
+    def format_response_payload(self, output: bytes) -> str:
+        response_json = json.loads(output)
+        return response_json[0]["0"]
+
+
+class HFContentFormatter(ContentFormatterBase[str, str]):
+    """Content handler for LLMs from the HuggingFace catalog."""
+
+    content_type = "application/json"
+    accepts = "application/json"
+
+    def format_request_payload(self, prompt: str, model_kwargs: Dict) -> bytes:
+        input_str = json.dumps({"inputs": [prompt], "parameters": model_kwargs})
+        return str.encode(input_str)
+
+    def format_response_payload(self, output: bytes) -> str:
+        response_json = json.loads(output)
+        return response_json[0][0]["generated_text"]
+
+
+class DollyContentFormatter(ContentFormatterBase[str, str]):
+    """Content handler for the Dolly-v2-12b model"""
+
+    content_type = "application/json"
+    accepts = "application/json"
+
+    def format_request_payload(self, prompt: str, model_kwargs: Dict) -> bytes:
+        input_str = json.dumps(
+            {"input_data": {"input_string": [prompt]}, "parameters": model_kwargs}
+        )
+        return str.encode(input_str)
+
+    def format_response_payload(self, output: bytes) -> str:
+        response_json = json.loads(output)
+        return response_json[0]
+
+
+class AzureMLOnlineEndpoint(LLM, BaseModel):
+    """Wrapper around Azure ML Hosted models using Managed Online Endpoints.
+
+    Example:
+        .. code-block:: python
+
+            auzre_llm = AzureMLModel(
+                endpoint_url="https://<your-endpoint>.<your_region>.inference.ml.azure.com/score",
+                endpoint_api_key="my-api-key",
+                deployment_name="my-deployment-name",
+                content_formatter=content_formatter)
+    """
+
+    endpoint_url: str = ""
+    """ URL of prexisting Endpoint """
+
+    endpoint_api_key: str = ""
+    """ Authentication Key for Endpoint"""
+
+    deployment_name: str = ""
+    """ Deployment Name for Endpoint"""
+
+    http_client: Any = None  #: :meta private:
+
+    content_formatter: Any = None
+    """The body handler class that provides an input and output
+    transform function to handle formats between the LLM and
+    the endpoint"""
+
+    """
```

Comment:
Wondering why do we have this block here? Is this placed correctly?

Reply:
I was doing something similar to the SageMaker endpoint, but it does not seem necessary to repeat the codeblock.